core/src/main/java/io/questdb/network/Socket.java

```diff
@@ -0,0 +1,114 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2023 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.network;
+
+import io.questdb.std.QuietCloseable;
+
+/**
+ * Abstraction for plain and encrypted TCP sockets. Encrypted sockets use additional buffer
+ * to accumulate messages, so they require extra calls to convert encrypted data to raw data.
+ */
+public interface Socket extends QuietCloseable {
```

Comment:
Add fake TLS socket implementation and some tests to make sure that `startTlsSession` gets called and nothing falls apart. That's to make ent more resistant to OSS contibutions.

Reply:
Added in b9f1d7d7ed24f43011bc8d5624764336d57044f0

cc @bluestreak01